packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -16,6 +128,18 @@ abstract class InAppPurchaseConnection {
   /// Query product details list that match the given set of identifiers.
   Future<ProductDetailsResponse> queryProductDetails(Set<String> identifiers);
 
+  /// Query all the past purchases.
+  ///
+  /// The `applicationUserName` is used for iOS only and it is optional. It does not have any effects on Android.
```

Comment:
Same thoughts above about asking people to case on this. Even if this isn't really used in Play I think it would probably be cleaner for users to just always pass this in if they also passed it in for originally making a purchase. I think it would be clearer here if we said that it is required in cases where this was passed in with the original transaction.